### DISCUSSION TITLE
PHP 8.3 with OC4

### DISCUSSION THREAD

**Discussion by TheCartpenter** [opencart/opencart#13128] [General]
PHP 8.3 has been released since almost a month. It would be great to create a compatibility for those series.

**Comment by mhcwebdesign** [comment]
Thanks for pointing it out.

I am going to test PHP 8.3 during the next few days.

**Comment by stalker780** [comment]
I am running PHP8.3 for several days and no problems so far
There are not much changes comparing 8.2 and 8.3

https://php.watch/versions/8.3

**Comment by TheCartpenter** [comment]
> I am running PHP8.3 for several days and no problems so far There are not much changes comparing 8.2 and 8.3
> 
> https://php.watch/versions/8.3

According to the PHP changelog, there are important changes to consider.

**Comment by stalker780** [comment]
> According to the PHP changelog, there are important changes to consider.

None that affect OC4 code

**Comment by stalker780** [comment]
Strange discussion about nothing. Just install PHP8.3 and check it with OC4 yourself. If you find anything to be fixed just make a PR.
How else you want to "create a compatibility" OC4 for PHP8.3?

**Comment by TheCartpenter** [comment]
> Strange discussion about nothing. Just install PHP8.3 and check it with OC4 yourself. If you find anything to be fixed just make a PR. How else you want to "create a compatibility" OC4 for PHP8.3?

There's nothing strange about being aware of a new PHP release in regard to changelog referring to methods and functions that the core already uses from the Engine and a bit from other locations. Yes, a PR will be created accordingly when issues will arise.

**Comment by AJenbo** [comment]
I have updated the tests to also include PHP 8.3. It did not find any new issues.
https://github.com/opencart/opencart/pull/13196

It is still a bit limited as PHPStan is only runs at level 0 (max is 9). One of my suggestions for you was that you could run it at higher levels to get a good list of issues to fix if you where interested. That will help solve bugs, prevent edge cases and make it much easier to make adjustments for new versions in the future.

**Comment by TheCartpenter** [comment]
Production servers will also provides their outputs, either from the front-end or the back-end, with future releases whether it originates from PHP or from OC, since each servers are configured differently from web hosting services on anyhow. That's why it's still good to address new releases when they arise.
